proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,168 @@
+# NuGetAudit without nuget.org as a package source
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Author: [zivkan](https://github.com/zivkan)
+- GitHub Issue: [12698](https://github.com/NuGet/Home/issues/12698)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+This proposal adds a new `<auditSource>` to *NuGet.Config* files, allowing *NuGetAudit* to obtain known vulnerabilities without adding nuget.org as a package source.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+[NuGetAudit](https://learn.microsoft.com/nuget/concepts/auditing-packages) is a feature where NuGet will report during restore which packages with known vulnerabilities are being used by projects, [proposed in late 2022](../2022/vulnerabilities-in-restore.md), and finally released with .NET 8 in 2023.
+The first version of NuGetAudit used package sources as the data source for known vulnerabilities, which will be referred to as a Vulnerability Database (VDB) for the remainder of this document.
+While nuget.org provides a VDB, there are several reasons why customers may not be using nuget.org as a package source.
+This is especially true following the 2021 blog post titled "Dependency confusion" which significantly increased knowledge of multi-source package substitution supply chain risks.
+
+While the design of NuGetAudit attempts to make it easy for upstreaming package sources to also upstream the `VulnerabilityInfo` resource, the NuGet team and customers are reliant of the feed implementor to implement the change.
+However, there are also cases where customers are not using upstreaming feeds, but instead curate approved 3rd party packages and manually upload them to an internal feed.
+What upstreaming means is a feed that is not solely a repository for packages that were pushed to the feed, but can also automatically obtain packages from other feeds.
+
+It's possible for customers to use [Package Source Mapping](https://learn.microsoft.com/nuget/consume-packages/package-source-mapping) to add nuget.org as a package source, but configured in a way that NuGet will never restore packages from this package source.
+However, some developers might consider this an unacceptable risk and prefer another way to get a VDB that cannot accidentally be used to get packages.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+*NuGet.Config* files have a new section `<auditSources>` where the URL(s) of any additional sources can be specified, which will be used only to download a VDB, and will not be used for downloading packages.
```

Comment:
> *NuGet.Config* files have a new section `<auditSources>` where the URL(s) of any additional sources can be specified

Would it be possible to download VDB from multiple audit sources? In other words, can customers configure multiple audit sources in the nuget.config file?

Reply:
Yes. I hoped that the example below that uses `<clear />` and `<add` made that sufficiently clear, despite not having an example with 2 audit sources. But I can add a sentence to explicitly state it.